src/Sylius/Bundle/ChannelBundle/Doctrine/ORM/ChannelRepository.php

```diff
@@ -16,28 +16,41 @@
 use Sylius\Bundle\ResourceBundle\Doctrine\ORM\EntityRepository;
 use Sylius\Component\Channel\Model\ChannelInterface;
 use Sylius\Component\Channel\Repository\ChannelRepositoryInterface;
+use Webmozart\Assert\Assert;
 
 class ChannelRepository extends EntityRepository implements ChannelRepositoryInterface
 {
     private const ORDER_BY = ['id' => 'ASC'];
 
     public function findOneByHostname(string $hostname): ?ChannelInterface
     {
-        return $this->findOneBy(['hostname' => $hostname], self::ORDER_BY);
+        $channel = $this->findOneBy(['hostname' => $hostname], self::ORDER_BY);
+        Assert::nullOrIsInstanceOf($channel, ChannelInterface::class);
```

Comment:
Cannot this be avoided with `phpstan-doctrine` extension?

It doesn't recognize return-type hint for magic methods?

Reply:
It doesn't work. Don't know, but probably the resource bundle repository in the middle break something on annotations.

Reply:
The `EntityRepository` from the `ResourceBundle` doesn't allow to pass a generic type 🫠.